core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -342,7 +342,7 @@ class LogSegment(val log: FileMessageSet,
    * The time this segment has waited to be rolled. If the first message in the segment does not have a timestamp,
    * the time is based on the create time of the segment. Otherwise the time is based on the timestamp of that message.
    */
-  def timeWaitedForRoll(now: Long) : Long= {
+  def timeWaitedForRoll(now: Long, messageTimestamp: Long) : Long= {
```

Comment:
We could do that. The only concern is that we have a debug level logging in the caller. Ideally we want to the same timestamp to be logged. Otherwise it may be confusing in some edge cases?